### PR TITLE
Feature/cvsb 3767

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: src/services/TestResultsService.js [hunk 1]
```diff
@@ -144,8 +144,9 @@ class TestResultsService {
         if (error.statusCode === 400 && error.message === 'The conditional request failed') {
           console.log('Error in insertTestResult > getTestTypesWithTestCodesAndClassification: Test Result id already exists', error)
           return Promise.reject(new HTTPResponse(201, 'Test Result id already exists'))
+        } else if (error.statusCode === 404 && error.body === 'No resources match the search criteria.') {
+          return Promise.reject(new HTTPResponse(404, 'Test types not found'))
         }
-        console.log('Error in insertTestResult > getTestTypesWithTestCodesAndClassification', error)
```
Comment (KHodderDVSA) [inline src/services/TestResultsService.js]: why was this removed? It was specifically added only a few weeks ago.

Comment (oalexandru) [reply src/services/TestResultsService.js]: By mistake. I've put it back

File: tests/unit/TestResultsService.unitTest.js [hunk 2]
```diff
@@ -309,6 +309,22 @@ describe('insertTestResult', () => {
...
+      const testResultsMockPostPayload = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../resources/test-results-post.json'), 'utf8'))
+      testResultsDAOMock.getTestCodesAndClassificationResponseFailFlag = true
+      testResultsDAOMock.testCodeAndClassificationResponse = new HTTPError(404, 'No resources match the search criteria.')
+      const testResultsService = new TestResultsService(testResultsDAOMock)
+
+      return testResultsService.insertTestResult(testResultsMockPostPayload[0])
+        .then(() => {})
+        .catch(error => {
+          expect(error.statusCode).to.equal(404)
+          expect(error.body).to.equal('"Test types not found"')
```
Comment (arupdvsa) [inline tests/unit/TestResultsService.unitTest.js]: please can these magic strings  '"Test types not found"' be replaced with enums
